### PR TITLE
Remove unused settings from db and codebase

### LINKED ISSUE #1966: Remove unused backend settings
## Problem Definition

The settings `thousand_separator`, `decimal_separator` and `currency_location` are only frontend settings. At some point in the past we stored them in the DB as separate settings though. This needs to be cleaned up

## Task

- [x] Remove all the backend code related to these specific settings.
- [x] Create a DB upgrade to remove these settings from the DB if they exist there already

### PR DESCRIPTION
**Pull request by vnavascues**
Migrates db from v22 to v23

I don't know if the frontend is still sending them. If so either we stop sending them or I modify `ModifiableSettingsSchema` for handling them as "deprecated" (although I haven't seen anything similar in `encoding.py`). Let me know!
 
Closes #1966

### CODE REVIEW

**Comment by LefterisJP** [comment]
Hmm @kelsos these settings were only frontend settings right? If so why did the e2e tests fail here? Is it unrelated?

File: rotkehlchen/db/upgrades/v22_v23.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+
+
+def upgrade_v22_to_v23(db: 'DBHandler') -> None:
```
Comment (LefterisJP) [inline rotkehlchen/db/upgrades/v22_v23.py:7]: Can you also add a DB upgrade test? It's easy.

You need to open the app with the previous DB version, create an empty account and populate those fields.

Take that DB and put it in the rest of the test DB data. Then use it in a DB upgrade test like done in the rest of the upgrades.

Essentially making sure that the settings are properly migrated to the frontend_settings key. And that the old settings are removed.

File: rotkehlchen/tests/db/test_db_upgrades.py [hunk 2]
```diff
@@ -1160,6 +1161,40 @@ def test_upgrade_db_21_to_22(user_data_dir):  # pylint: disable=unused-argument
     assert db.get_version() == 22
 
 
+def test_upgrade_db_22_to_23(user_data_dir):  # pylint: disable=unused-argument
```
Comment (LefterisJP) [inline rotkehlchen/tests/db/test_db_upgrades.py:1164]: Test is good! This is for the case the user has the settings populated in their DB. In the case it's missing we probably also need one as I just saw the upgrade code had an error which that test would catch.

Ofcourse upgrades happen only once in the lifetime of the app so the test won't be that useful after next version, but better have it to be on the safe side.

**Review by LefterisJP — COMMENTED** [review]
Looking good. Still two things left for this PR:

1. The frontend needs to also take care to now use the frontend_settings for these 3 settings. @kelsos 
2. The DB upgrade code needs to properly handle the case where the setting has not been populated in the DB (this is the most common case) @vnavascues

**Comment by LefterisJP** [comment]
hey @vnavascues so after https://github.com/rotki/rotki/pull/2014 I think it makes sense to add a cleaning of the bitfinex trades and used query ranges in this upgrade too.

I will also move 2014 in the bug fixes branch. If we make a patch release that part of the upgrade chould also go there but since the users can also clean the exchange history cache on their own it's not super critical I think.

File: rotkehlchen/tests/db/test_db_upgrades.py [hunk 3]
```diff
@@ -1161,14 +1161,41 @@ def test_upgrade_db_21_to_22(user_data_dir):  # pylint: disable=unused-argument
...
-    _use_prepared_db(user_data_dir, 'v22_rotkehlchen.db')
+    _use_prepared_db(user_data_dir, 'v22_rotkehlchen_w_frontend_settings.db')
+    db_v22 = _init_db_with_target_version(
+        target_version=22,
+        user_data_dir=user_data_dir,
+        msg_aggregator=msg_aggregator,
+    )
+    cursor = db_v22.conn.cursor()
+
+    # Check all settings exist
```
Comment (vnavascues) [inline rotkehlchen/tests/db/test_db_upgrades.py:1181]: Given the fact the db data before the migration is not visible at a first glance, and we have 2 db files with a subtle change, this is how I like to test it:
- Assert pre-conditions
- Migrate
- Assert changes

Let me know if you like it!

**Comment by vnavascues** [comment]
@LefterisJP all tests pass except e2e (as expected).

File: rotkehlchen/tests/db/test_db_upgrades.py (hunk 3 above)
Comment (LefterisJP) [reply rotkehlchen/tests/db/test_db_upgrades.py:1181]: Sure it's fine!

File: rotkehlchen/tests/db/test_db_upgrades.py [hunk 4]
```diff
@@ -1160,6 +1161,148 @@ def test_upgrade_db_21_to_22(user_data_dir):  # pylint: disable=unused-argument
...
+    # Check all settings except 'frontend_settings' exist
+    assert cursor.execute(
+        'SELECT COUNT(*) FROM settings WHERE name = "frontend_settings";',
+    ).fetchone()[0] == 0
+    assert cursor.execute(
+        'SELECT COUNT(*) FROM settings WHERE name IN '
+        '("thousand_separator", "decimal_separator", "currency_location");',
+    ).fetchone()[0] == 3
+
+    # Check Bitfinex trades and their used query range exist
```
Comment (LefterisJP) [inline rotkehlchen/tests/db/test_db_upgrades.py:1261]: No need for the bitfinex assertions in both tests. Just remove them.

File: rotkehlchen/tests/db/test_db_upgrades.py [hunk 5]
```diff
@@ -1160,6 +1161,148 @@ def test_upgrade_db_21_to_22(user_data_dir):  # pylint: disable=unused-argument
...
+    # Make sure the settings have been migrated into 'frontend_settings'
+    frontend_settings = cursor.execute(
+        'SELECT value FROM settings WHERE name = "frontend_settings";',
+    ).fetchone()[0]
+    frontend_settings_map = json.loads(frontend_settings)
+    assert frontend_settings_map['thousand_separator'] == ','
+    assert frontend_settings_map['decimal_separator'] == '.'
+    assert frontend_settings_map['currency_location'] == 'after'
+
+    # Make sure Bitfinex trades used query range has been deleted
```
Comment (LefterisJP) [inline rotkehlchen/tests/db/test_db_upgrades.py:1292]: no need for the bitfinex assertions in both tests. Just remove them.

**Review by LefterisJP — COMMENTED** [review]
@kelsos It seems to work, but the tax report period seem to have some problems:

1. It seems that it has preselected latest year and All. So I pressed generate and got an API error for `None` timestamp. When I clicked somewhere else and then clicked again to 2021 and all it worked. So I think the initial preselection has a problem. Error was

```
{"to_timestamp": ["Failed to deserialize a timestamp entry from string NaN"], "from_timestamp": ["Failed to deserialize a timestamp entry from string NaN"]}
```

2. The timestamps for the selection seem to not be in UTC. I selected Q4 of 2020 and it did 1601503200 to 1609455599 which is `09/30/2020 @ 10:00pm (UTC)` to `12/31/2020 @ 10:59pm (UTC)`. So Berlin timezone with DST on. I think this may need to become configurable an option, but by default we should use UTC.

For (2) it's tricky ... need to have consistent behaviour across all time selections. Does not need to be in this PR, but probably needs customization. There is no right way after thinking more about it.

File: rotkehlchen/tests/db/test_db_upgrades.py (hunk 4 above)
Comment (LefterisJP) [reply rotkehlchen/tests/db/test_db_upgrades.py:1261]: I will do so in my PR. No need to wait more for this.

**Review by LefterisJP — APPROVED** [review]

**Pull request merged**